### PR TITLE
Implements infestation

### PR DESCRIPTION
**Pull request by shayebeadling**

### CODE REVIEW

File: src/data/battle-anims.ts [hunk 1]
```diff
@@ -98,7 +98,8 @@ export enum CommonAnim {
     ELECTRIC_TERRAIN,
     GRASSY_TERRAIN,
     PSYCHIC_TERRAIN,
-    LOCK_ON = 2120
+    LOCK_ON = 2120,
+    INFESTATION
```
Comment (Flashfyre) [inline src/data/battle-anims.ts:102]: Please put this under an appropriate ID instead of appending

File: src/data/enums/battler-tag-type.ts [hunk 2]
```diff
@@ -49,5 +49,6 @@ export enum BattlerTagType {
   IGNORE_FLYING = "IGNORE_FLYING",
   SALT_CURED = "SALT_CURED",
   CHARGED = "CHARGED",
-  GROUNDED = "GROUNDED"
+  GROUNDED = "GROUNDED",
+  INFESTATION = "INFESTATION",
```
Comment (Flashfyre) [inline src/data/enums/battler-tag-type.ts:53]: Put this in an order that makes sense with the moves

**Review by Flashfyre — CHANGES_REQUESTED** [review]

File: src/data/battle-anims.ts (hunk 1 above)
Comment (shayebeadling) [reply src/data/battle-anims.ts:102]: If an id is provided, does it just use the normal move anim? If so, I can probably remove the copied common-infestation animation I created

Comment (shayebeadling) [reply src/data/battle-anims.ts]: ah I think I see what you meant, lemme know if the change I made isn't what you were talking about

**Review by Flashfyre — APPROVED** [review]

**Pull request merged**